### PR TITLE
DHFPROD-4440: Improving how resetProject works in HubTestBase

### PR DESCRIPTION
**Pull request by rjrudin**
Submitting this as part of DHFPROD-4440, which involves step processors. I started writing a JUnit test for this and realized that our tests can be sped up by not having to load hub artifacts all the time (because clearDatabases no longer needs them) and by not having to generate and deploy entity-based search options, which are never used. 

Main change is that clearDatabases in HubTestBase no longer deletes the hub-core-artifact collection. That keeps all DH core artifacts present. Had to improve some legacy tests that were assuming that the database was empty after calling clear (making assertions on the number of docs in a database leads to brittle tests).

Also made a tweak to LoadUserModulesCommand that both speeds up tests and fixes a long-standing issue when running "mlWatch", which is - when you run mlWatch, DHF generates and deploys entity-based search options over and over and over. That also adds some time to every test, and none of our tests depend on LoadUserModulesCommand deploying entity-based search options.

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub/src/test/java/com/marklogic/hub/AbstractHubTest.java [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
 
-import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 
 /**
- * Adding this so that a class can subclass HubTestBase without having to define the same two Spring annotations
+ * Adding this so that a class can subclass HubTestBase without hing to define the same two Spring annotations
```
Comment (srinathgit) [inline marklogic-data-hub/src/test/java/com/marklogic/hub/AbstractHubTest.java:9]: A small typo here. Other than that everything looks good.

Comment (rjrudin) [reply marklogic-data-hub/src/test/java/com/marklogic/hub/AbstractHubTest.java:9]: Thanks @srinathgit  - I'll have another PR for 4440 with the real impl in it; I'll fix the typo in that one.
